rbac/management/principal/it_service.py

```diff
@@ -303,8 +303,34 @@ def get_service_accounts(self, user: User, options: dict[str, Any] = {}) -> Tupl
         offset = options.get("offset")
         limit = options.get("limit")
         limit_offset_validation(offset, limit)
+        # Service account filtering query parameters
+        name = options.get("name")
+        owner = options.get("owner")
+        description = options.get("description")
+        sa_query_passed = name or owner or description
 
         count = len(service_accounts)
+        # If any one service account filter parameter is provided extract & return the specific service accounts
+        if sa_query_passed:
+            filtered_service_accounts = []
+            for sa in service_accounts:
+                sa_description = str(sa.get("description"))
+                if (
+                    (not name or sa.get("name") == name)
+                    and (not owner or sa.get("owner") == owner)
+                    and (not description or description in sa_description)
```

Comment:
I tried to use the filtering by description and I noticed that the filtering is case sensitive 🤔 and I believe we should have this case insensitive ... what you think @astrozzc

Reply:
Sounds good, let's use case insensitive then